MGMT-4013 Add generate-all Makefile target

YuviGold opened the pull request:
Move all the generation scripts from the Makefile into a bash script.
Including mocks, ocp_versions, swagger outputs, and manifests.

Run all of them by "generate-all".

CI would run all of them from 'ci-lint' Makefile target and fail in
case of any git changes that weren't committed.

Keeping the same API for all generations. `make generate-from-swagger` would be translated into `./hack/generate.sh generate_from_swagger`.
Except for mocks generation, it will become `make generate-mocks` instead of just `make generate`.

In case of an invalid generation operation, the next "help" message would be printed:
```
Function "generate_swagger" unavailable.
The available functions are:
	generate-all
	generate-from-swagger
	generate-go-client
	generate-go-server
	generate-keys
	generate-manifests
	generate-migration
	generate-mocks
	generate-ocp-version
	generate-python-client
make: *** [generate-swagger] Error 1
```

/cc @filanov  @nmagnezi @carbonin @osherdp

YuviGold commented on Makefile:
```diff
@@ -85,53 +86,8 @@ $(BUILD_FOLDER):
-validate-swagger-changes:
-	skipper make generate-from-swagger
-	git diff --exit-code client/ restapi/  # this will fail if changes to the client/server code have not been committed
+generate-%: ${BUILD_FOLDER}
```
Required fewer changes.
There's no added benefit for a Python script calling bash commands. And there wasn't a real logic over there.

filanov replied:
it's just more complicated to work with bash. but OK

YuviGold commented on Makefile:
```diff
@@ -68,13 +67,15 @@ endif
 REPORTS = $(ROOT_DIR)/reports
 TEST_PUBLISH_FLAGS = --junitfile-testsuite-name=relative --junitfile-testcase-classname=relative --junitfile $(REPORTS)/unittest.xml
 
+.EXPORT_ALL_VARIABLES:
```
https://www.gnu.org/software/make/manual/html_node/Special-Targets.html
```
.EXPORT_ALL_VARIABLES
Simply by being mentioned as a target, this tells make to export all variables to child processes by default. See Communicating Variables to a Sub-make.
```

Basically, that all the variables that are defined in the Makefile, will be available in the scripts that are being called.

filanov commented:
Few issues
1. No auto complete :(
2. the make targets are cont consistent `skipper make generate-from_swagger`

omertuc commented on Makefile:
```diff
@@ -68,13 +67,15 @@ endif
-	$(MAKE) verify-latest-onprem-config
-	$(MAKE) validate-swagger-changes
+	skipper $(MAKE) generate-all
+	git diff --exit-code  # this will fail if changes have not been committed
```
Great, maybe change the comment to `# this will fail if generate-all caused any diff`

YuviGold commented:
> Few issues
> 
> 1. No auto complete :(
> 2. the make targets are cont consistent `skipper make generate-from_swagger`
> 3. you get the usage only if you call `make generate-` not very friendly

@filanov 
1. Not at the moment. It would be hard to make it work unless we will deliver an "autocomplete" script to be sourced. Otherwise, there's no way to do it just by using the Makefile as is.
2. It's actually `generate-from-swagger` that translated into `generate_from_swagger`. I fixed the help message.
3. Added `make generate` to print the help message.
